.circleci/config.yml

```diff
@@ -26,18 +26,83 @@ jobs:
           paths:
             - node_modules
           key: v1-dependencies-{{ checksum "package.json" }}
-        
+
       - run:
           name: unit tests
           command: npm test
-          
+
       - run:
           name: coverage reporting
           command: npm run coverage:submit
 
-      - deploy:
-          name: publish
-          command: |
-            if [ "${CIRCLE_BRANCH}" == "master" ]; then
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - middy/*
+
+  build_next:
+    docker:
+      - image: circleci/node:8.10
+
+    working_directory: ~/middy
+
+    # These are almost the same set of steps as above.
+    # Yaml doesn't allow you to concatenate lists: https://github.com/yaml/yaml/issues/35
+    # Not in these steps:
+    #  - uploading code coverage
+    #  - saving workspace for the deploy job
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+          - v1-dependencies-{{ checksum "package.json" }}
+          - v1-dependencies-
+
+      - run:
+          name: configure NPM registry
+          command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
```

Comment:
Hm, if you don't publish in this step, then no need for `NPM_TOKEN` in `build_next ` job?